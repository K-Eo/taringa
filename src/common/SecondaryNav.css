.SecondaryNav .navbar-text.divider {
  background: rgba(0, 0, 0, 0.3);
  margin: 0 14px 0 18px;
  width: 1px;
}

.SecondaryNav .btn-light {
  color: rgba(0, 0, 0, 0.3);
}

.SecondaryNav .nav-link.active {
  color: rgba(0, 0, 0, 1) !important;
  font-weight: 500 !important;
}

.SecondaryNav .btn-light.active {
  background-color: transparent !important;
  border-color: transparent !important;
  color: #6cb7f4 !important;
}
