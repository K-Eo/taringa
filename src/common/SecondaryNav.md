Basic

```js
const { MemoryRouter } = require('react-router-dom');
<MemoryRouter>
  <SecondaryNav />
</MemoryRouter>
```
