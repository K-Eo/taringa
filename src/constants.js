export const REPLACE = 'REPLACE'
export const PUSH = 'PUSH'
