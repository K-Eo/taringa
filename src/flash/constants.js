export const ADD = 'flash/ADD'
