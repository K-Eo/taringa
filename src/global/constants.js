export const filters = {
  hot: {
    displayName: 'Destacados',
    exact: true,
    id: 'hot',
    pathname: '/global',
    url: '/feed/global',
  },
  recents: {
    displayName: 'Recientes',
    exact: false,
    id: 'recents',
    pathname: '/global/recents',
    url: '/feed/global?sort=recent',
  },
  tops: {
    displayName: 'Tops',
    exact: false,
    id: 'tops',
    pathname: '/global/tops',
    url: '/feed/global/tops/week',
  },
}
