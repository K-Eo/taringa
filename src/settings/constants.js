export const ITEM_BIG = 'settings/ITEM_BIG'
export const ITEM_MEDIUM = 'settings/ITEM_MEDIUM'
export const ITEM_SMALL = 'settings/ITEM_SMALL'

export const CHANGE_ITEM_SIZE = 'settings/CHANGE_ITEM_SIZE'
