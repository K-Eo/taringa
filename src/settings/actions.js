import { CHANGE_ITEM_SIZE, ITEM_MEDIUM } from './constants'

export const changeItemSize = size => ({
  type: CHANGE_ITEM_SIZE,
  size: size,
  storage: {
    key: 'type',
    value: 'size',
  },
})

export const restoreItemSize = store => {
  if (localStorage) {
    const size = localStorage.getItem(CHANGE_ITEM_SIZE) || ITEM_MEDIUM
    store.dispatch(changeItemSize(size))
  }
}
