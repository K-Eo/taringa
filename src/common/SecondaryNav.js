import React from 'react'
import { NavLink, withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import classNames from 'classnames'

import './SecondaryNav.css'
import { ITEM_BIG, ITEM_MEDIUM, ITEM_SMALL } from '../settings/constants'
import * as actions from '../settings/actions'

export const SecondaryNav = ({ itemSize, changeItemSize }) => {
  const bigButtonClass = classNames('btn px-1 py-0 btn-light', {
    active: itemSize === ITEM_BIG,
  })
  const mediumButtonClass = classNames('btn px-1 py-0 btn-light', {
    active: itemSize === ITEM_MEDIUM,
  })
  const smallButtonClass = classNames('btn px-1 py-0 btn-light', {
    active: itemSize === ITEM_SMALL,
  })

  return (
    <div className="SecondaryNav bg-light mb-4 border-bottom">
      <div className="container">
        <div className="row">
          <div className="col-8">
            <nav className="navbar navbar-light navbar-expand-lg p-0">
              <div className="collapse navbar-collapse">
                <div className="form-inline">
                  <button
                    className={bigButtonClass}
                    onClick={() => changeItemSize(ITEM_BIG)}
                  >
                    <i className="ta-items-big" />
                  </button>

                  <button
                    className={mediumButtonClass}
                    onClick={() => changeItemSize(ITEM_MEDIUM)}
                  >
                    <i className="ta-items-medium" />
                  </button>

                  <button
                    className={smallButtonClass}
                    onClick={() => changeItemSize(ITEM_SMALL)}
                  >
                    <i className="fa fa-align-justify" />
                  </button>
                </div>

                <span className="navbar-text divider" />

                <div className="navbar-nav mr-auto">
                  <NavLink
                    activeClassName="active"
                    className="nav-item nav-link"
                    exact
                    to="/"
                  >
                    Destacados
                  </NavLink>

                  <NavLink
                    activeClassName="active"
                    className="nav-item nav-link"
                    to="/recents"
                  >
                    Recientes
                  </NavLink>

                  <NavLink
                    activeClassName="active"
                    className="nav-item nav-link"
                    to="/tops"
                  >
                    Tops
                  </NavLink>
                </div>

                <div className="form-inline">
                  <button className="btn btn-sm btn-outline-success">
                    Crear Post
                  </button>
                </div>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = state => ({
  itemSize: state.settings.itemSize,
})

export default withRouter(connect(mapStateToProps, actions)(SecondaryNav))
