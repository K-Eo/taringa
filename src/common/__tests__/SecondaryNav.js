import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import { ITEM_BIG, ITEM_MEDIUM, ITEM_SMALL } from '../../settings/constants'
import { SecondaryNav } from '../SecondaryNav'

describe('Secondary Nav', () => {
  it('exists', () => {
    expect(SecondaryNav).toBeDefined()
  })

  it('renders', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SecondaryNav />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders active on ITEM_BIG', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SecondaryNav itemSize={ITEM_BIG} />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders active on ITEM_MEDIUM', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SecondaryNav itemSize={ITEM_MEDIUM} />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders active on ITEM_SMALL', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SecondaryNav itemSize={ITEM_SMALL} />
        </MemoryRouter>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('handles onClick on ITEM_BIG', () => {
    const mock = jest.fn()

    const wrapper = shallow(<SecondaryNav changeItemSize={mock} />)

    wrapper
      .find('button')
      .at(0)
      .simulate('click')
    expect(mock).toBeCalledWith(ITEM_BIG)
  })

  it('handles onClick on ITEM_MEDIUM', () => {
    const mock = jest.fn()

    const wrapper = shallow(<SecondaryNav changeItemSize={mock} />)

    wrapper
      .find('button')
      .at(1)
      .simulate('click')
    expect(mock).toBeCalledWith(ITEM_MEDIUM)
  })

  it('handles onClick on ITEM_SMALL', () => {
    const mock = jest.fn()

    const wrapper = shallow(<SecondaryNav changeItemSize={mock} />)

    wrapper
      .find('button')
      .at(2)
      .simulate('click')
    expect(mock).toBeCalledWith(ITEM_SMALL)
  })

  it('renders filters', () => {
    const filters = [
      {
        displayName: 'Foo',
        exact: true,
        id: 'foo',
        pathname: '/',
      },
      {
        displayName: 'Bar',
        exact: false,
        id: 'bar',
        pathname: '/bar',
      },
    ]

    const wrapper = shallow(<SecondaryNav filters={filters} />)

    expect(wrapper).toMatchSnapshot()
  })
})
