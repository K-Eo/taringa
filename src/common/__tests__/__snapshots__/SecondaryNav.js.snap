// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secondary Nav renders 1`] = `
<div
  className="SecondaryNav bg-light border-bottom"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-8"
      >
        <nav
          className="navbar navbar-light navbar-expand-lg p-0"
        >
          <div
            className="collapse navbar-collapse"
          >
            <div
              className="form-inline"
            >
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-big"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-medium"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="fa fa-align-justify"
                />
              </button>
            </div>
            <span
              className="navbar-text divider"
            />
            <div
              className="navbar-nav mr-auto"
            />
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`Secondary Nav renders active on ITEM_BIG 1`] = `
<div
  className="SecondaryNav bg-light border-bottom"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-8"
      >
        <nav
          className="navbar navbar-light navbar-expand-lg p-0"
        >
          <div
            className="collapse navbar-collapse"
          >
            <div
              className="form-inline"
            >
              <button
                className="btn px-1 py-0 btn-light active"
                onClick={[Function]}
              >
                <i
                  className="ta-items-big"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-medium"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="fa fa-align-justify"
                />
              </button>
            </div>
            <span
              className="navbar-text divider"
            />
            <div
              className="navbar-nav mr-auto"
            />
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`Secondary Nav renders active on ITEM_MEDIUM 1`] = `
<div
  className="SecondaryNav bg-light border-bottom"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-8"
      >
        <nav
          className="navbar navbar-light navbar-expand-lg p-0"
        >
          <div
            className="collapse navbar-collapse"
          >
            <div
              className="form-inline"
            >
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-big"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light active"
                onClick={[Function]}
              >
                <i
                  className="ta-items-medium"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="fa fa-align-justify"
                />
              </button>
            </div>
            <span
              className="navbar-text divider"
            />
            <div
              className="navbar-nav mr-auto"
            />
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`Secondary Nav renders active on ITEM_SMALL 1`] = `
<div
  className="SecondaryNav bg-light border-bottom"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-8"
      >
        <nav
          className="navbar navbar-light navbar-expand-lg p-0"
        >
          <div
            className="collapse navbar-collapse"
          >
            <div
              className="form-inline"
            >
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-big"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-medium"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light active"
                onClick={[Function]}
              >
                <i
                  className="fa fa-align-justify"
                />
              </button>
            </div>
            <span
              className="navbar-text divider"
            />
            <div
              className="navbar-nav mr-auto"
            />
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`Secondary Nav renders filters 1`] = `
<div
  className="SecondaryNav bg-light border-bottom"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-8"
      >
        <nav
          className="navbar navbar-light navbar-expand-lg p-0"
        >
          <div
            className="collapse navbar-collapse"
          >
            <div
              className="form-inline"
            >
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-big"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="ta-items-medium"
                />
              </button>
              <button
                className="btn px-1 py-0 btn-light"
                onClick={[Function]}
              >
                <i
                  className="fa fa-align-justify"
                />
              </button>
            </div>
            <span
              className="navbar-text divider"
            />
            <div
              className="navbar-nav mr-auto"
            >
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-item nav-link"
                exact={true}
                key="foo"
                to={
                  Object {
                    "pathname": "/",
                    "state": Object {
                      "clear": true,
                    },
                  }
                }
              >
                Foo
              </NavLink>
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-item nav-link"
                exact={false}
                key="bar"
                to={
                  Object {
                    "pathname": "/bar",
                    "state": Object {
                      "clear": true,
                    },
                  }
                }
              >
                Bar
              </NavLink>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
